mock/py/mock.py

```diff
@@ -666,6 +674,8 @@ def main():
     util.setup_host_resolv(config_opts)
 
     buildroot = Buildroot(config_opts, uidManager, state, plugins, bootstrap_buildroot)
+    signal.signal(signal.SIGTERM, partial(handle_sigterm, buildroot))
```

Comment:
Hmm, should we catch/process SIGABRT as well?

Reply:
On which occasions (other than internal error) does mock get `SIGABRT`?

I think it would make sense handle for example `SIGINT`, which is usually received when mock runs in the foreground process group and the user presses `^C`.

Reply:
E.g. `copr-rpmbuild` sends (on timeout) SIGINT down to mock as well.

Reply:
Well, SIGINT is interpreted as KeyInterruptError in python and is already caught. As I mentioned below after KeyInterruptError all processes are killed, but it is not my work. If someone would test this too, it will be better.

Reply:
Indeed, `SIGINT` seems to be handled in recent versions of mock.

What about `SIGHUP` and `SIGPIPE`?

Try the following:
1. run `mock --chroot 'echo sleeping... && sleep 1024' &`
2. wait till `sleeping...` appears in your terminal
3. run `kill -SIGHUP $!` or `kill -SIGPIPE $!`
4. run `ps -A | grep sleep`

Reply:
SIGHUP is a good idea.
SIGPIPE - I am not sure I want to encourage any use of pipes with mock. It has always been recommended to use logs with mock. Ẅith stdout, mock's code assumes human on the other side.

Reply:
But copr, koji and covscan are working with live logs...  which actually has potential to get broken pipes.

Reply:
That said - speaking about SIGPIPE - it would be nice to have at least some was to tell the mock that on stdout/stderr isn't consumed by human.